src/main/java/com/bookshop/controller/BookController.java

```diff
@@ -33,12 +34,14 @@ public class BookController {
     @GetMapping
     @Operation(summary = "Get all books by page",
             description = "Get a list of all books at x page with y size")
+    @PreAuthorize("hasAnyRole('USER', 'ADMIN')")

```

Comment:
shouldn't admin user also have a role `USER`? or you design your app in a way that admin does not have same priviliges as user has?